lib/membrane_native/raw_video.ex

```diff
@@ -0,0 +1,92 @@
+defmodule Membrane.Native.RawVideo do
+  @moduledoc """
+  Bundlex compatible implementation of the Membrane.RawVideo struct.
+
+  Currently bundlex does not support tuples in structs, so we have to decompose framerate into two fields.
+  """
+
+  @typedoc """
+  Currently supported formats used to encode the color of every pixel in each video frame.
+  """
+
+  @type pixel_format_t ::
+          :I420 | :I422 | :I444
+
+  @typedoc """
+  Width of single frame in pixels.
+  """
+  @type width_t :: pos_integer()
+
+  @typedoc """
+  Height of single frame in pixels.
+  """
+  @type height_t :: pos_integer()
+
+  @typedoc """
+  Numerator of number of frames per second. To avoid using tuple type,
+  it is described by 2 separate integers number.
+  """
+  @type framerate_num_t :: non_neg_integer
+
+  @typedoc """
+  Denominator of number of frames per second. To avoid using tuple type,
+  it is described by 2 separate integers number. Default value is 1.
+  """
+  @type framerate_den_t :: pos_integer
+
+  @type t :: %__MODULE__{
+          width: width_t(),
+          height: height_t(),
+          pixel_format: pixel_format_t(),
+          framerate_num: framerate_num_t(),
+          framerate_den: framerate_den_t(),
+          aligned: boolean()
+        }
+  @enforce_keys [:width, :height, :pixel_format, :framerate_num, :aligned]
+  defstruct width: nil,
+            height: nil,
+            pixel_format: nil,
+            framerate_num: nil,
+            framerate_den: 1,
+            aligned: nil
+
+  @supported_pixel_formats [:I420, :I422, :I444]
```

Comment:
It doesn't make sense here, there's no reason why the values should be limited vs RawVideo

Reply:
Some native implementations do not support all pixel formats. 
But it may be an additional argument in the conversion function.